service/lib/agama/storage/volume_conversion/from_y2storage.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/storage_manager"
+require "agama/storage/volume_templates_builder"
+
+module Agama
+  module Storage
+    module VolumeConversion
+      # Volume conversion from Y2Storage format.
+      class FromY2Storage
+        # @param spec [Y2Storage::VolumeSpecification]
+        # @param config [Agama::Config]
+        def initialize(spec, config:)
+          @spec = spec
+          @config = config
+        end
+
+        # Performs the conversion from Y2Storage format.
+        #
+        # @return [Agama::Storage::Volume]
+        def convert
+          volume = VolumeTemplatesBuilder.new_from_config(config).for(spec.mount_point || "")
+
+          volume.tap do |target|
+            target.device = spec.device
+            target.separate_vg_name = spec.separate_vg_name
+            target.mount_options = spec.mount_options
+            target.fs_type = spec.fs_type
+
+            sizes_conversion(target)
+            btrfs_conversion(target)
+          end
+        end
+
+      private
+
+        # @return [Y2Storage::VolumeSpecification]
+        attr_reader :spec
+
+        # @return [Agama::Config]
+        attr_reader :config
+
+        # @param target [Agama::Storage::Volume]
+        def sizes_conversion(target)
+          target.auto_size = !(spec.ignore_fallback_sizes? && spec.ignore_snapshots_sizes?)
+
+          planned = planned_device_for(spec.mount_point)
+          target.min_size = planned&.min || spec.min_size
+          target.max_size = planned&.max || spec.max_size
```

Comment:
to be honest I do not get what this code means. Why planned device min size has precedence over min size from specification? I understand code, but do not understand reason. It would be nice to have rationale explained in comment.

Reply:
Ok, added.